anax-in-k8s/cronjobs/auto-upgrade-cronjob.sh

```diff
@@ -245,91 +283,204 @@ function rollback_agent_image() {
+                    STATUS_PATH=$filepath/$nmp_name/status.json
+                fi
+            fi
+        fi
+    done
+
+    # Exit early if status.json does not exist
+    local no_active_jobs
+    test -f $STATUS_PATH 2>&1
+    no_active_jobs=$?
+    if [ "$no_active_jobs" -eq "1" ]; then
+        log_info "There are no active auto upgrade jobs."
+        write_logs
+        exit 0
+    fi
+
+    log_verbose "Found job: $STATUS_PATH"
+
+    log_debug "get_status_path() end"
+}
+
+function restart_agent_pod() {
+    log_debug "function restart_agent_pod() start"
+
+    log_info "Restarting agent pod..."
+    cmd_output=$( { $KUBECTL rollout restart deployment agent -n ${AGENT_NAMESPACE}; } 2>&1 )
+    rc=$?
+    if [[ $rc != 0 ]]; then
+        rollback_failed "There was an unexpected error while restarting agent pod: $cmd_output"
+    fi
+
+    log_debug "function restart_agent_pod() end"
+}
+
 #====================== Main  ======================
 
-# Exit early if status.json does not exist
-auto_upgrade_disabled=$(agent_cmd "test -f /var/horizon/status.json")
-if [ auto_upgrade_disabled ]; then
-    rollback_failed "The /var/horizon/status.json does not exist."
-fi
+# Sets STATUS_PATH for rest of script
+get_status_path
 
 # Check agent deployment/pod status and status.json
 pod_status=$($KUBECTL get pods ${POD_ID} --no-headers -o custom-columns=":status.phase")
 log_debug "Pod status: $pod_status"
 dep_status=$($KUBECTL rollout status deployment/agent -n ${AGENT_NAMESPACE} | awk '{ print $3 }' | sed 's/successfully/Running/g')
 log_debug "Deployment status: $dep_status"
-json_status=$(agent_cmd "cat /var/horizon/status.json | jq '.status' | sed 's/\"//g'")
+json_status=$(cat $STATUS_PATH | jq '.agentUpgradePolicyStatus.status' | sed 's/\"//g')
 log_debug "Cron Job status: $json_status"
 
-# Exit early if cron job already executed to success or failure
-if [[ "$json_status" == "rollback failed" ]]; then
-    rollback_failed "Rollback was already initiated, but it failed. Deployment status: $dep_status, Pod status: $pod_status"
-elif [[ "$json_status" == "rollback successful" ]]; then
-    log_info "Rollback already completed successfully."
-    exit 0
-fi
-
+# Check deployment/pod status
 log_info "Checking if agent is running and deployment is successful..."
 if [[ "$pod_status" != "Running" || "$dep_status" != "Running" ]]; then
     
-    # If k8s deployment status is "error" and if status.json has "rollback started" -> set status to "rollback failed" (stop here)
+    # Should never happen, but if k8s deployment status is "error" and if status.json has "rollback started", set status to "rollback failed" and exit
     log_info "Agent is not running. Checking if rollback was already attempted..."
-    if [[ "$json_status" == "rollback started" ]]; then
+    if [[ "$json_status" == "$STATUS_ROLLBACK_STARTED" ]]; then
         rollback_failed "Rollback was already initiated, but it failed. Deployment status: $dep_status"
```

Comment:
Can you wait $AGENT_DEPLOYMENT_STATUS_TIMEOUT_SECONDS (right now it sets to 75s) before calling `rollback_failed` function at here:
```
 if [[ "$json_status" == "$STATUS_ROLLBACK_STARTED" ]]; then
    wait for 75s
    rollback_failed "Rollback was already initiated, but it failed. Deployment status: $dep_status"
fi
```

Because  `pod/deploy status != Running` doesn't mean it fail, maybe it is just slow to bring up pod.

Reply:
That's a good catch. I'm adding that